pkg/collector/distinct_string_collector.go

```diff
@@ -76,22 +97,26 @@ func (d *DistinctString) Exceeded() bool {
 	d.mtx.Lock()
 	defer d.mtx.Unlock()
 
-	return d.totalLen > d.currLen
+	return d.limExceeded
 }
 
-// TotalDataSize is the total size of all distinct strings encountered.
-func (d *DistinctString) TotalDataSize() int {
+// Size is the total size of all distinct strings encountered.
+func (d *DistinctString) Size() int {
 	d.mtx.Lock()
 	defer d.mtx.Unlock()
 
-	return d.totalLen
+	return d.currLen
 }
 
 // Diff returns all new strings collected since the last time diff was called
-func (d *DistinctString) Diff() []string {
+func (d *DistinctString) Diff() ([]string, error) {
 	d.mtx.Lock()
 	defer d.mtx.Unlock()
 
+	if !d.diffEnabled {
+		return nil, errDiffNotEnabled
+	}
```

Comment:
This path doesn't require the lock, we can acquire it after

Reply:
yup, moved the lock down in all collectors.